### PR TITLE
AC-473: Update EncounterDAO for Room

### PR DESCRIPTION
**Pull request by Gelassen**
## Description of what I changed
Continue migration to Room.  Implement Ecnounter wrapper and cover its work with tests.

I left all old active directory classes as well to finish migration to Room first and only then replace active directory logic.

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-473

## Checklist: I completed these to help reviewers :)
- [  ] My pull request only contains **ONE single commit**
(my PR contains 2 commit because it depends on previous PR which hasn't been merged yet).

- [ x ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [ ] All new and existing **tests passed**.

- [ x ] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by HerbertYiga** [comment]
could you squash your commits

File: build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.3.2'
+        classpath 'com.android.tools.build:gradle:3.4.1'
```
Comment (f4ww4z) [inline build.gradle]: @Gelassen This is fixed. Please pull latest changes from master.

**Comment by dkayiwa** [comment]
@Gelassen are you still working on this?

**Comment by f4ww4z** [comment]
@Gelassen Ping me to reopen this when you resume working on it.

**Pull request closed without merging**